Add test for concurrent sweeps

raghavkarol opened the pull request:
Reintroduce and fix the test removed in 054b62c. This testcase
validates sweep concurrency functionality asserting that an expected
number of sweeps are running in concurrently.

Commit:
Add test for concurrent sweeps

Reintroduce and fix the test removed in 054b62c. This testcase
validates sweep concurrency functionality asserting that an expected
number of sweeps are running in concurrently.

Commit:
Remove unused function

raghavkarol commented on itest/riak_kv_sweeper_SUITE.erl:
```diff
@@ -985,25 +985,34 @@ exometer_deleted_object_count(Config) ->
                     {ok, [{count, DeleteKeys}, {one, DeleteKeys}]}),
     ok.
 
+scheduler_sweep_concurrency_test(_Config) ->
```
I will add checks to make sure the scheduler doesn't go over the limit also

Commit:
Use 1 as first index when creating partitions

Were creating Partitions+1 indices

Commit:
Update sweep concurrency test

Add checks that we do not start more than sweep_concurrency_limit
sweeps in parallel.

nickelization commented:
Test looks good, but I think we need to rebase this on top of the latest sweeper-2.3-integration changes.

Commit:
Update test to remove usage of #sweep record

Sweep record is now incapsulated in the state, see dd7c4b5.

nickelization commented:
:+1:

nickelization committed:
Fix sweep concurrency test for latest rec changes

Now that sweep_participant is encapsulated in riak_kv_sweeper_fold, this
test case needs to be updated to match.

nickelization commented:
Going to just go ahead and merge this. It's a PR on top of a PR to add a test case, Raghav and I have both looked at it, and the test case is passing.
